aws_lambda_powertools/utilities/data_classes/alb_event.py

```diff
@@ -1,4 +1,6 @@
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, MutableMapping
+
+from requests.structures import CaseInsensitiveDict
```

Comment:
I really like this idea, but I'm not sure we can do it. We have not included the `requests` library in our layer/package and this will force the client to add the requests library, which can lead to challenges about versioning, conflicts and others.

I created a Lambda, added the Powertools Layer and look at this error

```python
import json
from requests.structures import CaseInsensitiveDict

def lambda_handler(event, context):
    # TODO implement
    return {
        'statusCode': 200,
        'body': json.dumps('Hello from Lambda!')
    }

```

<img width="1307" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/4295173/bff925cf-271c-4a53-b49c-382a6edbec6b">

Reply:
Oh yeah, my fault! boto3 uses urllib3 directly, but not requests. Maybe we should replace requests' CaseInsensitiveDict by urllib3's [HTTPHeaderDict](https://github.com/urllib3/urllib3/blob/79a2a30f06b61127dc138158b82ffa791ab16733/src/urllib3/_collections.py#L206), also a subclass of `MutableMapping` (so my comment regarding `MutableMapping` still applies).

Reply:
@leandrodamascena, I just found an issue with urllib3's HTTPHeaderDict that might not serve all our requirements: the values can only be of type `str`. And we do have cases where values are of type `byte` (in KafkaEventRecord) or `List[str]` (multiValueHeaders).

Reply:
I'm thinking if we need to rely on some external dependencies or we can create our own `SomenameDict` class and implement the init, get, iter methods. We don't need to add or delete items as this is immutable. Does it make sense to you? If so, go ahead and let's focus on unblocking this item before we move on, okay? I'm focused on merging this PR by next week, so please let me know if I can help with anything.

Once again, thank you very much for your effort in working on such important PR.

Reply:
Cool, let me add our own implementation of `CaseInsensitiveDict` then. We still need the implementation to be mutable actually, exactly because of the _normalize_multi_header_values_with_param function.

Reply:
Tbh, I don't like the idea of adding this new dependency, and let me explain why:

1. The premise of the EventSource DataClass is to be a very fast way to transform JSON into an object using "pure" Python, with no dependencies required. If we bring in a new dependency, we will need to add it to our layer and pip package and force customers to install that, what is not ideal right now.

2. We do open source, and we love to adopt other open-source libraries, but we need to choose libraries very carefully. Supply chain is something we need to be mindful of every day, and I look at this library and see PRs opened since 2023 without a merge, and the last commit was 8 months ago. If we choose to adopt this library and, for some reason, the developers archive this project or even remove the PyPI packages, we are in trouble.

3 - This change is a must because we are wanting to reuse code, as today it's too verbose to extract headers. So, I think we can create a tiny class to do that with only the necessary methods we need, and then reuse this class in our code.

Make sense for you all these points?